drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenterTest.java

```diff
@@ -1641,4 +1655,17 @@ public void testRefreshView() {
         verify(parent).refreshScrollPosition();
         verify(layer).draw();
     }
+    
+    private void checkDTEventFired(int times){
+        final ArgumentCaptor<DecisionTableSelectionsChangedEvent> dtSelectionsChangeEventCaptor = ArgumentCaptor.forClass(DecisionTableSelectionsChangedEvent.class);
+ 
+        verify(decisionTableSelectionsChangedEvent,times(times)).fire(dtSelectionsChangeEventCaptor.capture());
+        
+        final DecisionTableSelectionsChangedEvent dtSelectionChangeEvent = dtSelectionsChangeEventCaptor.getValue();
```

Comment:
If the ```times``` is more than one there will be multiple values; therefore it's probably best to use ```dtSelectionsChangeEventCaptor.getAllValues()```, check the returned ```List``` size equals ```times``` and that each instance's ```presenter``` equals ```dtPresenter```

Reply:
@manstis thanks for clarifying that bit of logic.